substrate/frame/revive/src/wasm/runtime.rs

```diff
@@ -949,20 +979,54 @@ impl<'a, E: Ext, M: ?Sized + Memory<E::T>> Runtime<'a, E, M> {
 		} else {
 			self.ext.get_storage(&key)
 		};
+
 		if let Some(value) = outcome {
 			self.adjust_gas(charged, costs(value.len() as u32));
-			self.write_sandbox_output(
-				memory,
-				out_ptr,
-				out_len_ptr,
-				&value,
-				false,
-				already_charged,
-			)?;
-			Ok(ReturnErrorCode::Success)
+
+			match read_mode {
+				StorageReadMode::ZeroIfMissing => {
```

Comment:
Needs a different name since it also influences the result if the value is not missing. Maybe something like `EthSemantics`. Cause it also fixes the output size to 32 byte.

You should add the `out_len_ptr` to one of the `StorageReadMode` variants. Since it is ignored. This is better than passing a `SENTINEL` which is never read.